changelog.d/gh-7839.fixed

```diff
@@ -0,0 +1,2 @@
+Fix for very long runtimes that could happen due to one of our optimizations. We now detect when that might
```

Comment:
I'm nitpicking, but as a user I would find this changelog entry kind of confusing, and as a developer I would like to know what optimization was that. I think users are less interested in knowing that the long runtimes could happen because of an optimization. They are more interested in knowing _when_ (what kind of files/rules) would trigger this perf issue.

Reply:
Thanks for the feedback! I'm too lazy to make a PR to improve the changelog since I don't have a much better explanation 😅 but I'll keep that in mind for next time